yaks/lib/yaks/format/json_api.rb

```diff
@@ -20,28 +20,32 @@ def call(resource, _env = nil)
         end
         output[:included] = included if included.any?
         output[:meta] = resource[:meta] if resource[:meta]
+
         output
       end
 
       # @param [Yaks::Resource] resource
       # @return [Hash]
       def serialize_links(links)
-        links.inject({}) do |hash, link|
-          hash.update(link.rel => link.uri)
+        links.each_with_object({}) do |link, hash|
+          hash[link.rel] = link.uri if link.rel.instance_of?(Symbol)
```

Comment:
This seems too restrictive. A Symbol rel means the rel is an IANA/Microformats officially registered link relationship type. Other relationships should per RFC5988 be fully qualified URIs. JSON-API does not mention RFC5988, even though they use standard registered rels (self, next, related, ...). From skimming the spec I also don't see that they would disallow using a fully qualified URI as a relationship, so I would prefer we include those since it's the most general solution. If we must use short identifiers there's still a way around that.

What should happen on the Yaks side is that we start representing rels as their own objects, so you can get several "views", e.g. RFC5988 rel vs "short name". In the meanwhile you could check if a `rel_template` is configured, and try to turn URLs back into "short names". E.g

```
rel_template 'http://example.org/rels/{rel}'

Yaks::Resource::Link.new(rel: "http://example.org/rels/author", ...) # => use above to turn this back into "author"
```